setup.py

```diff
@@ -589,284 +596,300 @@ def get_blosc_version(headername):
     # Try to locate the compulsory and optional libraries.
     lzo2_enabled = False
     compiler = new_compiler()
-    for (package, location) in [(hdf5_package, HDF5_DIR),
-                                (lzo2_package, LZO_DIR),
-                                (lzo1_package, LZO_DIR),
-                                (bzip2_package, BZIP2_DIR),
-                                (blosc_package, BLOSC_DIR)]:
-
-        if package.tag == 'LZO' and lzo2_enabled:
-            print("* Skipping detection of %s since %s has already been found."
-                % (lzo1_package.name, lzo2_package.name))
+    for (package, location) in [
+        (hdf5_package, HDF5_DIR),
+        (lzo2_package, LZO_DIR),
+        (lzo1_package, LZO_DIR),
+        (bzip2_package, BZIP2_DIR),
+        (blosc_package, BLOSC_DIR),
+    ]:
+
+        if package.tag == "LZO" and lzo2_enabled:
+            print(
+                f"* Skipping detection of {lzo1_package.name} "
+                f"since {lzo2_package.name} has already been found."
+            )
             continue  # do not use LZO 1 if LZO 2 is available
 
         # if a package location is not specified, try to find it in conda env
         if not location and CONDA_PREFIX:
             location = CONDA_PREFIX
 
-        # looking for lzo/lzo1x.h but pkgconfig already returns '/usr/include/lzo'
-        use_pkgconfig = USE_PKGCONFIG if package.tag != 'LZO2' else False
+        # looking for lzo/lzo1x.h but pkgconfig already returns
+        # '/usr/include/lzo'
+        use_pkgconfig = USE_PKGCONFIG and package.tag != "LZO2"
 
         (hdrdir, libdir, rundir) = package.find_directories(
-            location, use_pkgconfig=use_pkgconfig)
+            location, use_pkgconfig=use_pkgconfig
+        )
 
         # check if HDF5 library uses old DLL naming scheme
-        if hdrdir and package.tag == 'HDF5':
-            hdf5_header = os.path.join(hdrdir, "H5public.h")
+        if hdrdir is not None and hdrdir.is_dir() and package.tag == "HDF5":
```

Comment:
Please check this (and the following ones).
If I remember correctly there is the possibility that ``hdrdir`` is ``True`` (see the implementation of [find_directories]( https://github.com/PyTables/PyTables/blob/cd26d6d88870a47e0231abb53105e08d292e3843/setup.py#L321)).

In this case
```
if hdrdir is not None and hdrdir.is_dir() and ...
```
is not equivalent to
```
if hdrdir and ...
``` 

Unfortunately I do not remember why this hack has been implemented.

Reply:
postponed to a new PR